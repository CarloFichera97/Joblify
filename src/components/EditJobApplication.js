import React from "react";
const EditJobApplication = () => <div> This is from Edit Job Application </div>;
export default EditJobApplication;
