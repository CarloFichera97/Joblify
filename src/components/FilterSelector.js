import React, { Component } from "react";
import { connect } from "react-redux";
import { DateRangePicker } from "react-dates";
import Select from "react-select";

const customStyles = {
  option: (provided, state) => ({
    ...provided,
  }),
  control: () => ({
    // none of react-select's styles are passed to <Control />
    width: 80,
    borderBottom: "1px solid #797a9e",
    padding: "3px",
    padding: "10px 10px 0px 0px",
    margin: "2px",
  }),

  dropdownIndicator: () => ({
    display: "none",
  }),

  singleValue: (provided, state) => {
    return { ...provided };
  },
};

const sortByOptions = [
  { value: "date", label: "Date" },
  { value: "salary", label: "Salary" },
];

const searchByOptions = [
  { value: "company", label: "Company" },
  { value: "role", label: "Role" },
];

import {
  sortByDate,
  sortBySalary,
  setCompanyName,
  setRoleName,
  searchByCompany,
  searchByRole,
  setEndDate,
  setStartDate,
} from "./../actions/filters";

class FilterSelector extends React.Component {
  state = {
    calendarFocused: null,
    selectedSortByOption: null,
    selectedSearchByOption: null,
  };

  onDatesChange = ({ startDate, endDate }) => {
    this.props.dispatch(setStartDate(startDate));
    this.props.dispatch(setEndDate(endDate));
  };

  handleSelectChangeSortBy = (selectedSortByOption) => {
    this.setState({ selectedSortByOption }, () => {
      if (this.state.selectedSortByOption.label === "Date") {
        this.props.dispatch(sortByDate());
      } else if (this.state.selectedSortByOption.label === "Salary") {
        this.props.dispatch(sortBySalary());
      }
    });
  };

  handleSelectChangeSearchBy = (selectedSearchByOption) => {
    this.setState({ selectedSearchByOption }, () => {
      if (this.state.selectedSearchByOption.label === "Company") {
        this.props.dispatch(searchByCompany());
      } else if (this.state.selectedSearchByOption.label === "Salary") {
        this.props.dispatch(searchByRole());
      }
    });
  };

  onFocusChange = (calendarFocused) => {
    this.setState(() => ({
      calendarFocused,
    }));
  };
  render() {
    return (
      <div className="content-container-filters">
        <div className="input-group">
          <div className="input-group__item">
            <p className="select_option"> Search By:</p>
            <Select
              isSearchable={false}
              styles={customStyles}
              options={searchByOptions}
              value={this.state.selectedSearchByOption}
              onChange={this.handleSelectChangeSearchBy}
              theme={(theme) => ({
                ...theme,
                colors: {
                  ...theme.colors,
                  text: "#797a9e",
                  primary25: "#b7b8cb",
                  primary: "#797a9e",
                },
              })}
            />
          </div>
          <div className="input-group__item">
            <p className="select_option"> Sort By:</p>

            <Select
              isSearchable={false}
              styles={customStyles}
              options={sortByOptions}
              value={this.state.selectedSortByOption}
              onChange={this.handleSelectChangeSortBy}
              theme={(theme) => ({
                ...theme,
                colors: {
                  ...theme.colors,
                  text: "#797a9e",
                  primary25: "#b7b8cb",
                  primary: "#797a9e",
                },
              })}
            />
          </div>
          <div className="input-group__item">
            <input
              placeholder="Insert Filter"
              className="dashboard_input"
              type="text"
              value={
                this.props.filters.searchBy === "company"
                  ? this.props.filters.company
                  : this.props.filters.role
              }
              onChange={(e) => {
                if (this.props.filters.searchBy === "company") {
                  this.props.dispatch(setCompanyName(e.target.value));
                } else if (this.props.filters.searchBy === "role") {
                  this.props.dispatch(setRoleName(e.target.value));
                }
              }}
            ></input>
          </div>

          <div className="input-group__item">
            <DateRangePicker
              startDate={this.props.filters.startDate}
              endDate={this.props.filters.endDate}
              onDatesChange={this.onDatesChange}
              focusedInput={this.state.calendarFocused}
              onFocusChange={this.onFocusChange}
              showClearDates={true}
              numberOfMonths={1}
              isOutsideRange={() => false}
              noBorder={true}
            />
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters,
  };
};

export default connect(mapStateToProps)(FilterSelector);
