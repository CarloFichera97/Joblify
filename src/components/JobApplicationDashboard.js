import React from "react";

const JobApplicationDashboard = () => <div> This is from the Dashboard</div>;

export default JobApplicationDashboard;
