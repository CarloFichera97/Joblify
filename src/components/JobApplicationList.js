import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import getFilteredJobApplications from "./../selectors/jobApplications";
import numeral from "numeral";
import "numeral/locales/en-gb";

export const JobApplicationList = (props) => {
  return (
    <div className="content-container__dashboard">
      <p className="numeral_locale">{numeral.locale("en-gb")}</p>

      <h2 className="total__filter__selected">
        There are a total of {props.jobApplications.length} Job Applications
        listed for the filters selected
      </h2>

      <div className="list-body">
        <div className="list-header">
          <div className="upper_message_company">Company & Role</div>
          <div className="upper_message_salary">Salary</div>
        </div>
        {props.jobApplications.map((jobApplication) => (
          <div key={jobApplication.id}>
            <Link className="list-item" to={`/view/${jobApplication.id}`}>
              <div>
                <p className="list-item__title">{jobApplication.company}</p>
                <span className="list-item__sub-title">
                  {" "}
                  {jobApplication.role}{" "}
                </span>
              </div>
              <span className="list-item__data">
                {numeral(jobApplication.salary / 100).format("$00,00.00")}{" "}
              </span>
            </Link>
          </div>
        ))}
      </div>
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    jobApplications: getFilteredJobApplications(
      state.jobApplications,
      state.filters
    ),
  };
};

export default connect(mapStateToProps)(JobApplicationList);
