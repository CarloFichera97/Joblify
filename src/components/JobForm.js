import React from "react";
import moment from "moment";
import { SingleDatePicker } from "react-dates";
import "react-dates/initialize";

export default class JobForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      company: props.jobApplication ? props.jobApplication.company : "",
      role: props.jobApplication ? props.jobApplication.role : "",
      salary: props.jobApplication ? props.jobApplication.salary / 100 : "",
      status: props.jobApplication ? props.jobApplication.status : "",
      description: props.jobApplication ? props.jobApplication.description : "",
      notes: props.jobApplication ? props.jobApplication.notes : "",
      createdOn: props.jobApplication
        ? moment(props.jobApplication.createdOn)
        : moment(),
      recruiterFullName: props.jobApplication
        ? props.jobApplication.recruiterFullName
        : "",
      recruiterEmail: props.jobApplication
        ? props.jobApplication.recruiterEmail
        : "",
      recruiterPhoneNumber: props.jobApplication
        ? props.jobApplication.recruiterPhoneNumber
        : "",
      calendarFocused: false,
      error: "",
    };
  }

  onCompanyChange = (e) => {
    const company = e.target.value;
    this.setState(() => ({ company }));
  };

  onRoleChange = (e) => {
    const role = e.target.value;
    this.setState(() => ({ role }));
  };
  onSalaryChange = (e) => {
    const salary = e.target.value;
    if (!salary || salary.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ salary }));
    }
  };
  onDescriptionChange = (e) => {
    const description = e.target.value;
    this.setState(() => ({ description }));
  };
  onStatusChange = (e) => {
    const status = e.target.value;
    this.setState(() => ({ status }));
  };

  onNotesChange = (e) => {
    const notes = e.target.value;
    this.setState(() => ({ notes }));
  };

  onNameChange = (e) => {
    const recruiterFullName = e.target.value;
    this.setState(() => ({ recruiterFullName }));
  };

  onEmailChange = (e) => {
    const recruiterEmail = e.target.value;
    this.setState(() => ({ recruiterEmail }));
  };

  onNumberChange = (e) => {
    const recruiterPhoneNumber = e.target.value;
    if (!recruiterPhoneNumber || recruiterPhoneNumber.match(/^\d{1,}?$/)) {
      this.setState(() => ({ recruiterPhoneNumber }));
    }
  };

  onDateChange = (createdOn) => {
    if (createdOn) {
      this.setState(() => ({
        createdOn,
      }));
    }
  };

  onFocusChange = ({ focused }) => {
    this.setState(() => ({
      calendarFocused: focused,
    }));
  };

  closeError = () => {
    this.setState(() => ({ error: "" }));
  };
  onSubmit = (e) => {
    e.preventDefault();
    if (
      !this.state.company ||
      !this.state.role ||
      !this.state.description ||
      !this.state.salary
    ) {
      this.setState(() => ({
        error:
          "Please provide at least Company Name, Role, Description and indicative salary",
      }));
    } else {
      this.setState(() => ({ error: "" }));
      this.props.onSubmit({
        company: this.state.company,
        role: this.state.role,
        salary: parseFloat(this.state.salary, 10) * 100,
        description: this.state.description,
        status: this.state.status,
        createdOn: this.state.createdOn.valueOf(),
        notes: this.state.notes,
        recruiterFullName: this.state.recruiterFullName,
        recruiterEmail: this.state.recruiterEmail,
        recruiterPhoneNumber: this.state.recruiterPhoneNumber,
      });
    }
  };

  render() {
    return (
      <div>
        {this.state.error && (
          <div onClick={this.closeError} className="error-header">
            <div className="content-container">
              <h1 className="page-header__title">
                {this.state.error && this.state.error}
              </h1>
            </div>
          </div>
        )}

        <form onSubmit={this.onSubmit}>
          <h1 className="title title-companydetails">Job Details</h1>

          <input
            className="input--jobform"
            type="text"
            placeholder="Company Name *"
            autoFocus
            value={this.state.company}
            onChange={this.onCompanyChange}
          />
          <input
            className="input--jobform"
            type="text"
            placeholder="Role  *"
            value={this.state.role}
            onChange={this.onRoleChange}
          />

          <SingleDatePicker
            date={this.state.createdOn}
            onDateChange={this.onDateChange}
            focused={this.state.calendarFocused}
            onFocusChange={this.onFocusChange}
            numberOfMonths={1}
            isOutsideRange={(day) => false}
            block
          />
          <input
            className="input--jobform"
            type="text"
            placeholder="Salary  *"
            value={this.state.salary}
            onChange={this.onSalaryChange}
          />
          <input
            className="input--jobform"
            type="text"
            placeholder="Description  *"
            value={this.state.description}
            onChange={this.onDescriptionChange}
          />
          <input
            className="input--jobform"
            type="text"
            placeholder="Status"
            value={this.state.status}
            onChange={this.onStatusChange}
          />
          <textarea
            className="textarea"
            rows="5"
            placeholder="Add a note for your job application"
            value={this.state.notes}
            onChange={this.onNotesChange}
          ></textarea>

          <h1 className="title title-recruiterdetails">Recruiter Details</h1>
          <input
            className="input--jobform"
            type="text"
            placeholder="Recruiter Name"
            value={this.state.recruiterFullName}
            onChange={this.onNameChange}
          />

          <input
            className="input--jobform"
            type="email"
            placeholder="Recruiter Email"
            value={this.state.recruiterEmail}
            onChange={this.onEmailChange}
          />

          <input
            className="input--jobform"
            type="text"
            placeholder="Recruiter Number"
            value={this.state.recruiterPhoneNumber}
            onChange={this.onNumberChange}
          />
          <div>
            <button className="button__jobForm" onClick>
              Add Job Application
            </button>
          </div>
        </form>
      </div>
    );
  }
}
