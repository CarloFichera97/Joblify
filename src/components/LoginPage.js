import React from "react";
import { connect } from "react-redux";
import { firebase, googleAuthProvider } from "../firebase/firebase";
const LoginPage = ({ startLogin }) => {
  return (
    <div>
      <button onClick={startLogin}>Login</button>
    </div>
  );
};
const startLogin = () => {
  firebase.auth().signInWithPopup(googleAuthProvider);
};

export const startLogout = () => {
  firebase.auth().signOut();
};

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin()),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
