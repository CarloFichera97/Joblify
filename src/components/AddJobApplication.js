import React from "react";
import { connect } from "react-redux";
import { startAddJobApplication } from "./../actions/jobApplications";
import JobForm from "./JobForm";

export class AddJobApplication extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      message: "",
    };
  }

  onSubmit = (jobApplication) => {
    this.props.startAddJobApplication(jobApplication);
    this.props.history.push("/dashboard");
  };

  render() {
    return (
      <div>
        <div className="content-container">
          <JobForm onSubmit={this.onSubmit} />
        </div>
      </div>
    );
  }
}

//To simplify testing with spies
const mapDispatchToProps = (dispatch) => ({
  startAddJobApplication: (jobApplication) =>
    dispatch(startAddJobApplication(jobApplication)),
});

export default connect(undefined, mapDispatchToProps)(AddJobApplication);
