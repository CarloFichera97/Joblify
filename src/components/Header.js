import React from "react";
import { useState } from "react";
import styled from "styled-components";
import { NavLink } from "react-router-dom";
import { startLogout } from "./LoginPage";

const NavigationBar = styled.nav`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 20px;
  min-height: 9vh;
  background: #354152;
  margin-bottom: 20px;
`;

const Title = styled.h1`
  font-size: 46px;
  color: white;
  font-style: italic;
  margin-left: 50px;
`;

const HorizontalMenu = styled.ul`
  list-style: none;
  display: flex;

  li:nth-child(2) {
    margin: 0px 21px;
  }

  @media (max-width: 700px) {
    display: none;
  }
`;

const ToggleIcon = styled.button`
  background: none;
  cursor: pointer;
  border: none;
  outline: none;

  @media (min-width: 700px) {
    display: none;
  }
`;

const LineHorizontalButton = styled.span`
  display: block;
  border-radius: 50px;
  width: 25px;
  height: 3px;
  margin: 5px;
  background-color: #fff;
  transition: width 0.4s ease-in-out;

  :nth-child(2) {
    width: ${(props) => (props.open ? "30%" : "70%")};
  }
`;

const Overlay = styled.div`
  position: absolute;
  height: ${(props) => (props.open ? "91vh" : 0)};
  width: 100%;
  background: #1c2022;
  transition: height 0.4s ease-in-out;
  @media (min-width: 700px) {
    display: none;
  }
`;

const OverlayMenu = styled.ul`
  list-style: none;
  position: absolute;
  left: 50%;
  top: 45%;
  transform: translate(-50%, -50%);

  li {
    opacity: ${(props) => (props.open ? 0.5 : 0)};
    font-size: 25px;
    margin: 50px 0px;
    transition: opacity 0.4s ease-in-out;
  }
`;

const Header = () => {
  const [toggle, toggleNav] = useState(false);
  return (
    <>
      <NavigationBar>
        <Title>Joblify</Title>
        <HorizontalMenu>
          <li>
            <NavLink
              to="/dashboard"
              activeClassName="
    active"
              className="nav_inactive_menu"
              exact={true}
            >
              Dashboard
            </NavLink>
          </li>

          <li>
            <NavLink
              to="/create"
              className="nav_inactive_menu"
              activeClassName="active"
            >
              Create
            </NavLink>
          </li>

          <li>
            <NavLink
              to="/help"
              className="nav_inactive_menu"
              activeClassName="active"
            >
              Help
            </NavLink>
          </li>

          <li>
            <a onClick={startLogout}>Logout</a>
          </li>
        </HorizontalMenu>
        <ToggleIcon onClick={() => toggleNav(!toggle)}>
          <LineHorizontalButton open={toggle} />
          <LineHorizontalButton open={toggle} />
          <LineHorizontalButton open={toggle} />
        </ToggleIcon>
      </NavigationBar>
      <Overlay open={toggle}>
        <OverlayMenu open={toggle}>
          <li>
            <NavLink
              onClick={() => {
                toggleNav(!toggle);
              }}
              to="/dashboard"
              exact={true}
            >
              Dashboard
            </NavLink>
          </li>

          <li>
            <NavLink
              onClick={() => {
                toggleNav(!toggle);
              }}
              to="/create"
            >
              Create
            </NavLink>
          </li>

          <li>
            <NavLink
              onClick={() => {
                toggleNav(!toggle);
              }}
              to="/help"
            >
              Help
            </NavLink>
          </li>

          <li>
            <a onClick={startLogout}>Logout</a>
          </li>
        </OverlayMenu>
      </Overlay>
    </>
  );
};

export default Header;
