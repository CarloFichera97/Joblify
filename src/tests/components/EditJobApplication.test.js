import React from "react";
import moment from "moment";
import { shallow } from "enzyme";
import { EditJobApplication } from "./../../components/EditJobApplication";
import { testDataArray, testData } from "./../fixtures/testData";

let onSubmit, history, onClick, wrapper;
beforeEach(() => {
  onSubmit = jest.fn();
  history = { push: jest.fn() };
  onClick = jest.fn();
  wrapper = shallow(
    <EditJobApplication
      onSubmit={onSubmit}
      history={history}
      onClick={onClick}
    />
  );
});

test("Should render EditJobApplication Correctly", () => {
  expect(wrapper).toMatchSnapshot();
});

test("Should handle onSubmit ", () => {
  wrapper.find("JobForm").prop("onSubmit")(testDataArray[0]);
  expect(history.push).toHaveBeenLastCalledWith("/");
  expect(onSubmit).toHaveBeenLastCalledWith(testDataArray[0]);
  expect(wrapper).toMatchSnapshot();
});

test("onClick called", () => {
  const wrapper = shallow(<button onClick={onClick} />);
  wrapper.find("button").simulate("click", "");
  expect(onClick.mock.calls.length).toEqual(1);
});

test("Should handle onClick ", () => {
  const wrapper = shallow(<button onClick={onClick} />);
  wrapper.find("button").prop("onClick")(testDataArray[0].id);
  expect(onClick).toHaveBeenLastCalledWith(testDataArray[0].id);
  expect(wrapper).toMatchSnapshot();
});
