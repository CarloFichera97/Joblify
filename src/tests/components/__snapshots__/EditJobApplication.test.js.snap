// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should handle onClick  1`] = `
<button
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [
          1,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
/>
`;

exports[`Should handle onSubmit  1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
        Edit Job Application
      </h1>
    </div>
  </div>
  <div
    className="content-container"
  >
    <JobForm
      onSubmit={[Function]}
    />
  </div>
</div>
`;

exports[`Should render EditJobApplication Correctly 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h1
        className="page-header__title"
      >
        Edit Job Application
      </h1>
    </div>
  </div>
  <div
    className="content-container"
  >
    <JobForm
      onSubmit={[Function]}
    />
  </div>
</div>
`;
